packages/vscode-extension/src/debugging/RadonCDPProxyDelegate.ts

```diff
@@ -0,0 +1,244 @@
+        })
+        .then((result): IProtocolSuccess => ({ id: reply.id, result }))
+        .catch((error): IProtocolError => ({ id: reply.id, error }));
+    }
+    return reply;
+  }
+
+  public async handleApplicationReply(
+    reply: IProtocolSuccess | IProtocolError,
+    tunnel: ProxyTunnel
+  ): Promise<IProtocolSuccess | IProtocolError | undefined> {
+    if (reply.id && "result" in reply && this.setBreakpointCommands.has(reply.id)) {
+      const finalReply = await this.maybeRetrySetBreakpointByUrl(reply, tunnel);
+      return finalReply;
+    }
+    return reply;
+  }
+
+  public async handleDebuggerReply(
+    reply: IProtocolSuccess | IProtocolError
+  ): Promise<IProtocolSuccess | IProtocolError | undefined> {
+    return reply;
+  }
+
+  private async onRuntimeEnable(tunnel: ProxyTunnel) {
+    await tunnel
+      .injectDebuggerCommand({
+        method: "FuseboxClient.setClientMetadata",
+        params: {},
+      })
+      .catch(_.noop);
+    await tunnel
+      .injectDebuggerCommand({
+        method: "ReactNativeApplication.enable",
+        params: {},
+      })
+      .catch(_.noop);
+  }
+
+  private handleScriptParsed(command: IProtocolCommand): IProtocolCommand {
+    const { sourceMapURL, url, scriptId } = command.params as Cdp.Debugger.ScriptParsedEvent;
+    if (!sourceMapURL) {
+      return command;
+    }
+
+    if (!sourceMapURL.startsWith("data:")) {
+      Logger.error(
+        "Source map URL doesn't encode source map data, mapping sources may not work correctly",
+        sourceMapURL
+      );
+      return command;
+    }
+
+    const base64Data = sourceMapURL.split(",")[1];
+    const decodedData = Buffer.from(base64Data, "base64").toString("utf-8");
+    const sourceMapData = JSON.parse(decodedData);
+
+    const isMainBundle = sourceMapData.sources.some((source: string) =>
+      source.includes("__prelude__")
+    );
+
+    this.sourceMapRegistry.registerSourceMap(sourceMapData, url, scriptId, isMainBundle);
+    return command;
+  }
+
+  private handleConsoleAPICalled(command: IProtocolCommand): IProtocolCommand | IProtocolSuccess {
+    const { args, stackTrace } = command.params as Cdp.Runtime.ConsoleAPICalledEvent;
+
+    // We wrap console calls and add stack information as last three arguments, however
+    // some logs may baypass that, especially when printed in initialization phase, so we
+    // need to detect whether the wrapper has added the stack info or not
+    // We check if there are more than 3 arguments, and if the last one is a number
+    // We filter out logs that start with __RNIDE_INTERNAL as those are messages
+    // used by IDE for tracking the app state and should not appear in the VSCode
+    // console.
+    if (args.length > 0 && args[0].value === "__RNIDE_INTERNAL") {
+      // We return here to avoid passing internal logs to the user debug console,
+      // but they will still be visible in metro log feed.
+      return { id: command.id!, result: {} };
```

Comment:
will this result in an empty log being appended to the vscode debug console?

Reply:
No, this sends a response to the application, it doesn't forward the command to the debugger.